app/models/course-leaderboard-entry.ts

```diff
@@ -14,11 +15,7 @@ export default class CourseLeaderboardEntry extends Model {
   @belongsTo('user', { async: false, inverse: null }) declare user: UserModel;
 
   get completedStagesCount() {
-    if (this.statusIsCompleted) {
-      return this.currentCourseStage.position;
-    } else {
-      return this.currentCourseStage.position - 1;
-    }
+    return this.completedStageSlugs?.length;
```

Comment:
```suggestion
    return this.completedStageSlugs.length;
```

(Given the type is `string[]`, we shouldn't need this)